### DISCUSSION TITLE
Timer issue with pico w

### DISCUSSION THREAD

**Discussion by ebolisa** [micropython/micropython#10657] [RP2040 / Pico]
Hi,

The code below fails with the msg: 

```
Traceback (most recent call last):
  File "<stdin>", line 12, in <module>
OSError: [Errno 12] ENOMEM
```
What am I doing wrong?

TIA
MicroPython v1.19.1-852-g9ea64a36a on 2023-02-03; Raspberry Pi Pico W with RP2040

```
from umachine import Timer # tried machine also

tim1 = Timer(-1) # tried Timer() also
tim2 = Timer(-1)

def test1():
    print('ok1')
    
def test2():
    print('ok2')

tim1.init(period=1000, mode=Timer.PERIODIC, callback=lambda t:test1())   #initializing the timer 1
tim2.init(period=2000, mode=Timer.PERIODIC, callback=lambda t:test2())   #initializing the timer 2
```

**Comment by peterhinch** [comment]
The above code works fine here.

**Reply by ebolisa** [reply]
Thanks.

**Comment by redhead-p** [comment]
Same here.  Ok on pico w and same MP release and on Arduino RP2040 connect with 1.19.1

**Reply by ebolisa** [reply]
Thanks.

**Comment by robert-hh** [comment]
Does it fail immediately or after a while?

**Reply by ebolisa** [reply]
Immediately. I know the code is good as it works on ESPs. I don't have an extra board to try with but will keep digging maybe by changing the CPU frequency.
